app/javascript/components/MainContainer/MainContainer.js

```diff
@@ -3,79 +3,88 @@ import RestaurantsList from "./RestaurantsList/RestaurantsList";
 import Filters from "./Filters/Filters"
 import Map from "./Map/Map"
 import './main_container.css'
+import { connect } from "react-redux";
+import { restListResponse, restChangeFilter } from "../actions/restaurantsActions";
+
 const axios = require('axios');
 
 
+@connect((store) => {
+    return {
+        restaurants: store.restaurantsList
+    };
+})
 class MainContainer extends Component {
     constructor(props) {
         super(props);
-        this.state = {
-            rating_filter: 'all',
-            cuisine_filter: 'all',
-            ten_bis_filter: 'all',
-            search_text: '',
-            restaurants: [],
-            filtered_restaurants: [],
-            cuisines: []
-        };
-
-        this.onChangeFilter = this.onChangeFilter.bind(this);
-        this.onChangeSearch = this.onChangeSearch.bind(this);
-        this.filterRestaurants = this.filterRestaurants.bind(this);
-
     }
 
     componentDidMount(){
+        this.reloadRestaurants();
+    }
+
+    reloadRestaurants = () => {
         axios.get('/restaurants.json')
             .then((response) => {return response.data;})
             .then((data) => {
                 let cuisines = [...new Set(data.map(x=>x.cuisine.toLowerCase()))].sort();
                 cuisines.unshift('all');
-                this.setState({ restaurants: data, filtered_restaurants: data, cuisines: cuisines });
+                this.props.dispatch(restListResponse(data, data, cuisines));
```

Comment:
I recommend you learn about `mapDispatchToProps`, it's the second parameter that goes into Redux's `connect`.
It binds between `dispatch` and your action creator function, giving you a proper dispatching function as a prop.